pkg/apis/service/v1alpha1/servicebinding_lifecycle.go

```diff
@@ -46,62 +47,50 @@ func (b *ServiceBinding) Do(ctx context.Context, ps *v1.WithPod) {
 	}
 	// TODO ensure unique volume names
 	// TODO limit volume name length
-	metadataVolume := fmt.Sprintf("%s-metadata", sb.Metadata.Name)
-	secretVolume := fmt.Sprintf("%s-secret", sb.Secret.Name)
-	if !existingVolumes.Has(metadataVolume) {
+	bindingVolume := fmt.Sprintf("%s-binding", sb.Name)
+	if !existingVolumes.Has(bindingVolume) {
 		ps.Spec.Template.Spec.Volumes = append(ps.Spec.Template.Spec.Volumes, corev1.Volume{
-			Name: metadataVolume,
+			Name: bindingVolume,
 			VolumeSource: corev1.VolumeSource{
 				ConfigMap: &corev1.ConfigMapVolumeSource{
-					LocalObjectReference: sb.Metadata,
+					LocalObjectReference: *sb,
 				},
```

Comment:
Shouldn't this rather be a `SecretVolumeSource`?
```go
Secret: &corev1.SecretVolumeSource{
        SecretName: sb.Name,
},
```

Reply:
yep